plugin/trino-queryeventlistener/src/main/java/io/trino/plugin/queryeventlistener/model/CreatedEvent.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.queryeventlistener.model;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.trino.spi.eventlistener.QueryCreatedEvent;
+
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+
+public class CreatedEvent
+{
+    private static final String KEY_EVENT_TYPE = "eventType";
+    private static final String KEY_QUERY_ID = "queryId";
+    private static final String KEY_CREATE_TIME = "createTime";
+    private static final String KEY_USER = "user";
+    private static final String KEY_SCHEMA = "schema";
+    private static final String KEY_CATALOG = "catalog";
+    private static final String KEY_SQL = "sql";
+    private static final String KEY_SOURCE = "source";
+    private static final String KEY_PRINCIPAL = "principal";
+
+    @JsonProperty(KEY_EVENT_TYPE)
+    private final String eventType = "QueryCreated";
+
+    @JsonProperty(KEY_QUERY_ID)
+    private final String queryId;
+
+    @JsonProperty(KEY_CREATE_TIME)
+    private final LocalDateTime createTime;
+
+    @JsonProperty(KEY_USER)
+    private final String user;
+
+    @JsonProperty(KEY_SCHEMA)
+    private final String schema;
+
+    @JsonProperty(KEY_CATALOG)
+    private final String catalog;
+
+    @JsonProperty(KEY_SQL)
+    private final String sql;
```

Comment:
Just for readability, since sql can be a massive string, can we make it the last property? I think putting principal right after user, and source after that would make a lot of sense.